### PR TITLE
test: listen, cancel and listen again

### PR DESCRIPTION
**Pull request by Vinzent03**
I've updated the StreamBuilder to be able to be listened again after subscription cancel.

### CODE REVIEW

**Comment by dshukertjr** [comment]
@Vinzent03  Thanks for this PR!

Is this PR related to this issue?
https://github.com/supabase-community/supabase-flutter/issues/219#issuecomment-1256232326

**Comment by Vinzent03** [comment]
Yeah kinda. I think you should be able to cancel listening and Realisten without recalling .stream.

**Review by dshukertjr — COMMENTED** [review]
I see. This PR looks good, but could we hold off on merging it until we find out the root cause within realtime-dart?

**Comment by Vinzent03** [comment]
Sure, this was my intention. I should have marked this as draft or mention in explicitly.
But I'm correct that it should re-request the server on a new channel subscription, right? I haven't understood the whole realtime architecture yet.

File: lib/src/supabase_stream_builder.dart [hunk 1]
```diff
@@ -29,7 +29,13 @@ typedef SupabaseStreamEvent = List<Map<String, dynamic>>;
...
   final PostgrestQueryBuilder _queryBuilder;
 
-  final RealtimeChannel _channel;
+  final RealtimeClient _realtimeClient;
+
+  final String _realtimeTopic;
+
+  RealtimeChannel? _channel;
+
+  PostgrestBuilder? _postgrestBuilder;
```
Comment (dshukertjr) [inline lib/src/supabase_stream_builder.dart:38]: Could we remove `_queryBuilder` and just use `_postgrestBuilder` here?

**Comment by dshukertjr** [comment]
>But I'm correct that it should re-request the server on a new channel subscription, right?

Yup, that should be correct as far as I know.

**Comment by Vinzent03** [comment]
@dshukertjr Turns out the initial connection from `RealtimeClient` is preserved. The issue were the wrong `ref` in the mocked server.

**Comment by Vinzent03** [comment]
@dshukertjr Do we want to include this in supabase-dart 1.0?

**Review by dshukertjr — APPROVED** [review]
Sorry, I missed your message from 10 days ago saying this one is complete. Certainly let's include it in the v1.0 update!

**Pull request merged**